File: 0797-all-paths-from-source-to-target/0797-all-paths-from-source-to-target.java
class Solution {
    List<List<Integer>> res = new ArrayList<>();
    
    private void dfs (int[][] graph, int idx, List<Integer> current) {

        if(idx == graph.length - 1) {            
            res.add(new ArrayList<>(current));
            return;
        }
        
        for(int i : graph[idx]) {
            current.add(i);
            dfs(graph,i,current);
            current.remove(current.size() - 1);
        }
    }
    
    public List<List<Integer>> allPathsSourceTarget(int[][] graph) {
        List<Integer> current = new ArrayList<>();
        current.add(0);
        this.dfs(graph,0,current);
        
        return res;
    }
}

File: 0797-all-paths-from-source-to-target/0797-all-paths-from-source-to-target.js
/**
 * @param {number[][]} graph
 * @return {number[][]}
 */
var allPathsSourceTarget = function(graph) {
    const [source,target] = [0,graph.length-1];

    const res = [];
    const dfs = (node,path) => {
        if(node === target) {
            res.push([...path]);
            path = [];
            return;
        }

        for(const neighbor of graph[node]) {
            path.push(neighbor);
            dfs(neighbor,path)
            path.pop();
        }
    }

    dfs(source,[0]);

    return res;
};